### DISCUSSION TITLE
Losing session in production

### DISCUSSION THREAD

**Discussion by irt-msummers** [etherCorps/SK-Redis-SessionManager#7] [General]
Have you seen where the session is lost when using a production version for node on Linux.  I have been able to deploy to vercel -- works fine.  I have a Windows :( deployment of node and it works fine -- but I deployed to AWS linux and it seems to create the session but when running the build version -- its like the session changes 

```
1421ba4b30ac7e93639805fc6c13cfca
Successfully created new session.
true
Unable to find data for the provided key - i2eV1iG8yRHoLvFThE7Z7NFuOfHMdIHUi1CE

```
but if I run the vite dev version -- it works fine
```
c34bcff5855ff7fe5830b6c3f4f04317
Successfully created new session.
false
Session Data
```

```
export const handle: Handle = async ({ event, resolve }) => {
	const userSession = await sessionManager.getSession(await event.cookies);

	event.locals = {
		isUserLoggedIn: false,
		user: null
	};
	console.log(userSession.error)
	console.log(userSession.message)
	if (userSession.error) {
		await sessionManager.deleteCookie(await event.cookies);
		return resolve(event);
	}
	if (userSession && userSession.data) {
		event.locals = {
			isUserLoggedIn: true,
			user: userSession?.data
		};
	}
	return resolve(event);
};

```

Is there anything in the sessionOptions that I need to set to get this to work

This is node v16.14.2 which is the same on mac which works fine

@sentry/node     ^7.84.0  
 @sentry/svelte   ^7.84.0  
 @sveltejs/kit    ^1.27.6 
 @types/node     ^20.10.1 
 typescript        ^5.3.2  
 vite              ^4.5.0  

Thanks

**Comment by theetherGit** [comment]
Which redis package?? ioredis, redis or upstash??
If possible can you share something so i can reproduce this?

**Comment by irt-msummers** [comment]
I found it. It was ioredis --but not the problem
The problem was the cookie was not even getting created in the first place. https://stackoverflow.com/questions/75045021/cookies-are-not-getting-set-when-running-with-host-option-in-official-svelteki
I don't know why it worked on Windows 
I just added secure: false to the cookiesOptions 

Thanks

**Reply by theetherGit** [reply]
Ok that's weird. If possible please share code snippet i wanna look what i can do for long term.

**Reply by irt-msummers** [reply]
In session.ts -- The strange thing is like I said it worked on my mac, worked on windows (node build) and vercel.  But the node build on linux had the issue (ran fine in testing on linux)
```
import {
	IoRedisSessionStore,
	type ioRedisSessionOptions
} from '@ethercorps/sveltekit-redis-session';


import { SECRET, REDIS_URL } from '$env/static/private';
import Redis from 'ioredis';

const sessionOptions: ioRedisSessionOptions = {
	redisClient: new Redis(REDIS_URL),
	secret: SECRET,
	sessionPrefix: 's-ibid',
	userSessionsPrefix: 'u-ibid',
	cookieName: 'session',
	cookiesOptions: {
yesterday

[Fix for cookie]
		maxAge: 60 * 60 * 24,
		// -- Change this at some point did for Linux to work
		secure: false


	}
};
console.log('The RedisURL is: ' +  REDIS_URL)
console.log('The RedisURL is: ' +  SECRET)
export const sessionManager = new IoRedisSessionStore(sessionOptions);
console.log('After Call to SessionManager')
```

**Reply by theetherGit** [reply]
Hard to think because vercel or all deploy services uses linux.

**Reply by irt-msummers** [reply]
Vercel uses SSL so I assume thats why it works -- it on 80 or anything else

**Reply by theetherGit** [reply]
Got it, You are not using SSL on linux server so it's not gonna apply secure cookies that's why it's not working. 
https (443) -> Secure Protocol -> Secure Cookies -> True
http (89) -> Unsecured Protocol -> Secure Cookies -> False

If you wanna use secure cookies than use SSL.
